modules/ingestor/src/service/advisory/csaf/creator.rs

```diff
@@ -37,17 +38,22 @@ impl PackageStatusCreator {
     pub fn add_all(&mut self, csaf: &Csaf, ps: &Option<Vec<ProductIdT>>, status: &'static str) {
         for r in ps.iter().flatten() {
             for purl in resolve_purls(csaf, r) {
-                let package = Purl::from(purl.clone());
+                let mut package = Purl::from(purl.clone());
+                package.qualifiers.clear();
 
                 if let Some(version) = package.version.clone() {
-                    self.entries.push(PackageStatus {
+                    let status = PackageStatus {
                         package,
                         status,
                         info: VersionInfo {
                             scheme: "generic".to_string(),
                             spec: VersionSpec::Exact(version),
                         },
-                    });
+                    };
+
+                    if !self.entries.contains(&status) {
```

Comment:
Wouldn't it make sense to use a `HashSet` for this? To me it looks like as if all struct could implement `Hash` and `Eq`.

Reply:
Good idea. I'll give that a whirl.